zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/oauth/OAuthCredentialsCache.java

```diff
@@ -45,38 +46,40 @@ public final class OAuthCredentialsCache {
       new TypeReference<Map<String, OAuthCachedCredentials>>() {};
   private static final ObjectMapper MAPPER = new ObjectMapper(new YAMLFactory());
 
-  private final Map<String, OAuthCachedCredentials> audiences;
   private final File cacheFile;
+  private final AtomicReference<Map<String, OAuthCachedCredentials>> audiences;
```

Comment:
❓ I'm curious why did we chose wrapping the map in an atomic reference, rather than for example using a `ConcurrentHashMap`?

`ConcurrentHashMap` supports atomic single operations on an `entry` level, rather than an `entire map` level with `AtomicReference `. It's also optimized for frequent read/write operations. Uses less memory (no need to copy the whole map with each update).

I can see that `AtomicReference<HashMap<>>` would be more suitable for cases with rare updates. 

But I'm curious to hear what was your reasoning  🙏

Reply:
@mustafadagher I understand you question and can't answer it for you. But since @korthout is on FTO this afternoon and the retreat is next week, do you mind if I go ahead and merge this without the answer? This way we can unblock the customer by giving him a snapshot version to verify against.

Reply:
I initially looked into using a concurrent hashmap but noticed specific access patterns like the clear and put all that's now become a reference set. AFAIK a clear and put all cannot be done atomically with a concurrent hashmap

Reply:
@remcowesterhoud, I've pre-approved the PR. you can go ahead and merged :)

@korthout I see! Thanks for answering! Indeed there's not an out-of-the-box atomic `clear` and `putAll` in `ConcurrentHashMap` without locking or synchronizing.

Reply:
Just a tiny hint. this change doesn't ensure an atomic single (entry level) put or get on the wrapped map AFAICT. To so, we might need to wrap a `ConcurrentHashMap` within an `AtomicReference`